ci/tasks.py

```diff
@@ -0,0 +1,196 @@
+    app=None,
+    outfile=None,
+):
+    full_diff = ""
+
+    release: Release
+    for release in c.calitp_config.channels[channel].releases:
+        if release.driver == ReleaseDriver.kustomize and (
+            not app or app == release.name
+        ):
+            kustomize_dir = c.calitp_config.git_root / Path(release.kustomize_dir)
+            cmd = f"kubectl diff -k {kustomize_dir}"
+            print(cmd, flush=True)
+            result: Result = c.run(
+                cmd,
+                warn=True,
+            )
+            if result.exited != 0:
+                full_diff += result.stdout
+    c.update({"kdiff": full_diff})
+    if outfile:
+        msg = f"```{full_diff}```" if full_diff else "No kustomize changes found.\n"
+        print(f"writing {len(msg)=} to {outfile}", flush=True)
+        with open(outfile, "w") as f:
+            f.write(msg)
+
+
+@task(parse_calitp_config)
+def hdiff(
+    c,
+    channel,
+    app=None,
+    outfile=None,
+):
+    full_diff = ""
+
+    release: Release
+    for release in c.calitp_config.channels[channel].releases:
+        if release.driver == ReleaseDriver.helm and (not app or app == release.name):
+            chart_path = c.calitp_config.git_root / Path(release.helm_chart)
+            cmd = f"helm dependency update {chart_path}"
+            print(cmd, flush=True)
+            c.run(
+                cmd,
+                warn=True,
+            )
+            with tempfile.TemporaryDirectory() as tmpdir:
+                manifest_path = Path(tmpdir) / Path("manifest.yaml")
+                kustomization_path = Path(tmpdir) / Path("kustomization.yaml")
+                values_str = " ".join(
+                    [
+                        f"--values {c.calitp_config.git_root / Path(values_file)}"
+                        for values_file in release.helm_values
+                    ]
+                )
+                cmd = f"helm template {release.helm_name} {chart_path} --namespace {release.namespace} {values_str} > {manifest_path}"
+                print(cmd, flush=True)
+                c.run(
+                    cmd,
+                    warn=True,
+                )
+                with open(kustomization_path, "w") as f:
+                    f.write(KUSTOMIZE_HELM_TEMPLATE.format(namespace=release.namespace))
+                cmd = f"kubectl diff -k {tmpdir}"
+                print(cmd, flush=True)
+                result: Result = c.run(
+                    cmd,
+                    warn=True,
+                )
+            if result.exited != 0:
+                full_diff += result.stdout
+    c.update({"hdiff": full_diff})
+    if outfile:
+        msg = f"```{full_diff}```" if full_diff else "No helm changes found.\n"
+        print(f"writing {len(msg)=} to {outfile}", flush=True)
+        with open(outfile, "w") as f:
+            f.write(msg)
+
+
+# TODO: we may want to split up channels into separate files so channel is not an argument but a config file
```

Comment:
+1

Reply:
I'll do that; I'd gone back-and-forth on everything in `invoke.yaml` vs providing configs, and I'm realizing that repetitive arguments in task definitions are kinda annoying. [pre/post-task data passing](https://github.com/pyinvoke/invoke/issues/261) is an open issue, so I think relying on config + context is the best pattern for now anyways.